### PR TITLE
Disable byteswap support on x86 until issues are resolved

### PR DESCRIPTION
**Pull request by fjeremic**
There are several issues encountered in the OpenJ9 project with the recent enablement of byteswap evaluators on x86 and Power. Some of the issues are explained in #5642. In this PR we fix one issue on Power, but there are still issues remaining which I will work with @aviansie-ben to fix. On x86 we need an overhaul of the 32-bit implementations and to take a look at the 64-bit failures as well.

### CODE REVIEW

**Comment by fjeremic** [comment]
@aviansie-ben could you please review? I'll touch base with you online to try and debug the last issue on Power which seems to happen even after disabling byteswap evaluators. Not sure how exactly yet but we will figure it out and get this re-enabled at least on Power.

File: compiler/p/codegen/OMRInstOpCodeProperties.hpp [hunk 1]
```diff
@@ -4770,7 +4770,7 @@
    /* .description =    "Store doubleword byte reverse indexed", */
    /* .prefix      = */ 0x00000000,
    /* .opcode      = */ 0x7C000528,
-   /* .format      = */ FORMAT_UNKNOWN,
+   /* .format      = */ FORMAT_RS_RA_RB_MEM,
```
Comment (aviansie-ben) [inline compiler/p/codegen/OMRInstOpCodeProperties.hpp:4773]: Well, it's a bit embarrassing that I missed this in the original move to the new binary encoder. :sweat_smile:

That being said, I generally have a policy of not defining formats for any new instructions here without adding corresponding tests to `fvtest/compilerunittest/p/BinaryEncoder.cpp` to ensure that the binary encoder unit tests remain complete.

**Review by aviansie-ben — CHANGES_REQUESTED** [review]

File: compiler/p/codegen/OMRInstOpCodeProperties.hpp (hunk 1 above)
Comment (fjeremic) [reply compiler/p/codegen/OMRInstOpCodeProperties.hpp:4773]: Fixed in 815ba08

**Comment by fjeremic** [comment]
@genie-omr build all

**Review by andrewcraik — APPROVED** [review]
I approve the x86-32 change as an interim functional measure. I leave the rest of the changes to someone more familiar with the p codegen

File: fvtest/compilerunittest/p/BinaryEncoder.cpp [hunk 2]
```diff
@@ -4355,6 +4355,10 @@ INSTANTIATE_TEST_CASE_P(StoreIndex, PPCMemSrc1EncodingTest, ::testing::ValuesIn(
     std::make_tuple(TR::InstOpCode::stwbrx,  MemoryReference(TR::RealRegister::gr31, TR::RealRegister::gr0 ), TR::RealRegister::gr0,   0x7c1f052cu, false),
     std::make_tuple(TR::InstOpCode::stwbrx,  MemoryReference(TR::RealRegister::gr0,  TR::RealRegister::gr31), TR::RealRegister::gr0,   0x7c00fd2cu, false),
     std::make_tuple(TR::InstOpCode::stwbrx,  MemoryReference(TR::RealRegister::gr0,  TR::RealRegister::gr0 ), TR::RealRegister::gr31,  0x7fe0052cu, false),
+    std::make_tuple(TR::InstOpCode::stdbrx,  MemoryReference(TR::RealRegister::gr0,  TR::RealRegister::gr0 ), TR::RealRegister::gr0,   0x7c00050cu, false),
+    std::make_tuple(TR::InstOpCode::stdbrx,  MemoryReference(TR::RealRegister::gr31, TR::RealRegister::gr0 ), TR::RealRegister::gr0,   0x7c1f050cu, false),
+    std::make_tuple(TR::InstOpCode::stdbrx,  MemoryReference(TR::RealRegister::gr0,  TR::RealRegister::gr31), TR::RealRegister::gr0,   0x7c00fd0cu, false),
+    std::make_tuple(TR::InstOpCode::stdbrx,  MemoryReference(TR::RealRegister::gr0,  TR::RealRegister::gr0 ), TR::RealRegister::gr31,  0x7fe0050cu, false),
```
Comment (aviansie-ben) [inline fvtest/compilerunittest/p/BinaryEncoder.cpp:4361]: The expected results for these tests look incorrect to me... According to the ISA document and output from gcc, it seems to me that `stdbrx 0, 0, 0` should be encoding as `0x7c000528u`.

**Review by aviansie-ben — CHANGES_REQUESTED** [review]

File: fvtest/compilerunittest/p/BinaryEncoder.cpp (hunk 2 above)
Comment (fjeremic) [reply fvtest/compilerunittest/p/BinaryEncoder.cpp]: Fixed in 6fddaab. I've disassembled online via [1].

Input:
```
7c 00 05 28
7c 1f 05 28
7c 00 fd 28
7f e0 05 28
```

Output:
```
.data:00000000 7c000528        stdbrx r0,0,r0
.data:00000004 7c1f0528        stdbrx r0,r31,r0
.data:00000008 7c00fd28        stdbrx r0,0,r31
.data:0000000c 7fe00528        stdbrx r31,0,r0
```

[1] https://onlinedisassembler.com/odaweb/

**Comment by fjeremic** [comment]
@genie-omr build all

**Review by aviansie-ben — APPROVED** [review]

**Pull request merged**